capture/scopes/scope.py

```diff
@@ -84,10 +85,18 @@ def _init_scope(self):
             scope.configure_batch_mode()
             return scope
         elif self.scope_cfg.scope_type == "waverunner":
-            # TODO WaveRunner needs to be adapted to new cycle-based config.
-            # (Issue lowrisc/ot-sca#210)
             if self.scope_cfg.ip:
                 scope = WaveRunner(self.scope_cfg.ip)
+
+                # Get current sampling rate from scope compare to cfg for sanity
+                setup_data = scope._ask("PANEL_SETUP?")
+                scope._get_and_print_cmd_error()
+                tmp_sampling_rate = int(re.findall(r'SampleRate = \d+', setup_data)[0][13:])
+                if self.scope_cfg.sampling_rate is not None:
+                    if self.scope_cfg.sampling_rate != tmp_sampling_rate:
+                        raise RuntimeError("WAVERUNNER: Error: WaveRunner sampling "
+                                           "rate does not match given configuration!")
```

Comment:
yeah it's not nice for the reason I stated in he above comment.

I am fine with different ways to deal with it:
- We could have two alternative scope constructors on the 'API', one based on samples, one based on cycles and internally they can nicely share the code.
- We can also remove the cycle cfg for waverunner alltogether.
- others ...

The only one I am opposed to is setting the sampling rate through config and loading it to the scope because it will mess up with the other settings very often and unpleasantly surprise lab technicians :D